app/src/main/java/com/github/se/polyfit/ui/components/selector/PictureSelector.kt

```diff
@@ -1,30 +1,92 @@
 package com.github.se.polyfit.ui.components.selector
 
+import android.Manifest
+import android.content.pm.PackageManager
+import android.graphics.Bitmap
+import android.widget.Toast
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.foundation.Image
 import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
 import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.layout.fillMaxWidth
 import androidx.compose.foundation.layout.size
 import androidx.compose.foundation.shape.RoundedCornerShape
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.unit.dp
+import androidx.core.content.ContextCompat
 import com.github.se.polyfit.ui.theme.PurpleGrey80
+import com.github.se.polyfit.viewmodel.post.CreatePostViewModel
+
+private const val PERMISSION_DENIED_MESSAGE = "Permission denied"
 
 @Composable
-fun PictureSelector(modifier: Modifier = Modifier) {
-  // Not sure if we are going to use a picture for the post, so this is just a placeholder
+fun PictureSelector(modifier: Modifier = Modifier, postViewModel: CreatePostViewModel) {
```

Comment:
Perhaps instead of passing the whole viewmodel, we can just pass `getBitMap` and `setBitMap`

Reply:
Yep did not think of that. That is way cleaner. Thanks for the advice !